### ISSUE TITLE
Benchmarks containing randomItem steps fail to parse

### ISSUE THREAD

**Issue by johnaohara** [Hyperfoil/Hyperfoil#163]
Benchmarks containing `randomItem` steps fail to parse with the following message;

```
ERROR: line 42, column 11: Failed to instantiate service-loaded builder randomItem: No builder implementing interface io.hyperfoil.api.config.StepBuilder with @Name randomItem
Failed to parse the benchmark.
```
To reproduce;

1. Startup hyperfoil in standalone mode from latest release
2. Start cli and connect to running instance
3. upload benchmark e.g. `upload examples/choose-movie.hf.yaml`

**Comment by rvansa** [comment]
I see that in your case it looks like a race condition; I cannot reproduce it. What are the involved threads that would invoke concurrent operations?

**Comment by johnaohara** [comment]
I did not manage to get the IDE to hit the breakpoint, idk if there is an IDE bug.   The Map is populated at `BUILDERS.computeIfAbsent(clazz, ServiceLoadedBuilderProvider::scanBuilders)` it appears to be a different thread, I suspect the JDK is using the forkJoin pool.

I can add some tracing to see which thread is populating the HashMap

**Comment by johnaohara** [comment]
I spent more time this weekend looking at the issue.  It is a classloader issue, not a thread safety issue. There appears to be a bug in the way UrlClassPath scans for classloaders. What is happening in "broken" case is the UrlClassPath finds the hyperfoil-http.jar before hyperfoil-core.jar, reads the meta-inf/manifest, which lists hyperfoil-core.jar on its classpath. UrlClassPath has an internal loaded map (lmap) and stores a null loader for hyperfoil-core.jar. when it actually finds hyperfoil-core-.jar it i thinks it already has a loader for hyperfoil-core, so ignores it and the loader is null, and therefore the ServiceLoader does not find all the services in hyperfoil-core.jar.  I thought it was a visibility issue as I couldn't set a breakpoint inside the ServiceLoader.load(builderInfo.implClazz).stream().forEach(provider -> { lambda in ServiceLoadedBuilderProvider and it appeared that Map<Class<?>, Map<String, BuilderInfo<?>>> BUILDERS was being populated in a different thread. Turns out it was an intellij issue (i have had a few) not stopping at breakpoints
me adding the the concurrentHashMap appeared to work, but it works correctly if i build from src from the release-0.14 tag
so.. i have been looking at the classloader issue, but I am having trouble with intellij again not stopping at breakpoints. Atm, this looks like a local environment issue with the classloader in the JDK, I am going to try a different IDE to see if I can get something that will let me debug this.
